main.tf

```diff
@@ -23,16 +23,18 @@ module "batch_subnet" {
 module "batch_queue" {
   source                = "./terraform/modules/swipe-sfn-batch-queue"
   app_name              = var.app_name
-  mock                  = var.mock
   batch_ssh_key_pair_id = length(aws_key_pair.swipe_batch) > 0 ? aws_key_pair.swipe_batch[0].id : ""
   network_info = length(module.batch_subnet) == 0 ? var.network_info : {
     vpc_id           = module.batch_subnet[0].vpc_id
     batch_subnet_ids = module.batch_subnet[0].batch_subnet_ids
   }
   ami_id                   = var.batch_ami_id
+  ami_ssm_parameter        = var.ami_ssm_parameter
+  miniwdl_dir              = var.miniwdl_dir
   batch_ec2_instance_types = var.batch_ec2_instance_types
   spot_min_vcpus           = var.spot_min_vcpus
   on_demand_min_vcpus      = var.on_demand_min_vcpus
+  cluster_types            = var.cluster_types
```

Comment:
Same here - also a replacement for `mock` where we can specify a list of the cluster types to launch for swipe.

Reply:
This is my only real sticking point with the PR. I think this parameter is kind of confusing and the interface isn't 100% defined. This can't really take an array of strings it must always be two strings and they must be either `EC2` or `SPOT`. It also only makes sense to change this for testing purposes because it only retries if it fails due to instance termination so the `EC2` `SPOT` configuration is the only sensible one.

I request you do one of the following:
- Use [custom-validation-rules](https://www.terraform.io/language/values/variables#custom-validation-rules) to enforce that only `["EC2", "EC2"]` and `["SPOT", "EC2"]` are allowed
- Replace with a boolean `use_spot` variable
- Stick with `mock` for this one

Reply:
I switched to a `use_spot` variable!